R/pension_supplement.R

```diff
@@ -0,0 +1,103 @@
+#' Can be claimed by those receiving Age Pension, Carer Payment, Wife Pension, Widow B Pension, Bereavement Allowance, or Disability Support Pension (except if under 21 and have no children).
+#' Can also be claimed if over age pension age and are receiving ABSTUDY, Austudy, Parenting Payment, Partner Allowance, Special Benefit, or Widow Allowance.
+#' Can still claim the basic amount if single, under age pension age, and receive the Parenting Payment.
+#'
+#' @param has_partner Does the individual have a partner?
+#' @param age The individual's age. Default is 18 years.
+#' @param n_dependants How many dependant children does the individual have?
+#' @param parenting_payment Is the individual receiving parenting payment?
+#' @param Date Date. Default is "2016/03/01" if fy.year is not present.
+#' @param fy.year Financial year. Default is "2015-16" if Date is not present.
+#' @param age_pension_age_requirement Does the payment for which the pension supplement is being added to require the individual to be over age pension age to receive? i.e. one of ABSTUDY, Austudy, Parenting Payment, Partner Allowance, Special Benefit, Widow Allowance. 
+#' @param disability_support_pension Is the individual receiving the disability support pension?
+#' @param per How often the payment will be made. Default is fortnightly. 
+#' @param overseas_absence Will the individual be living outside of Australia for more than 6 weeks of the year?
+#' @param seperated_couple Is the individual part of an illness separated couple, respite care couple, or partner imprisoned?
+#' 
+#' @author Matthew Katzen
+#' @export
+
+#note: annual = rate multiplied by 364/14 (= 26)
+
+pension_supplement <- function(has_partner = FALSE,
+                               age = 70,
+                               n_dependants = 0,
+                               parenting_payment = FALSE,
+                               Date = NULL,
+                               fy.year = NULL,
+                               age_pension_age_requirement = FALSE,
+                               disability_support_pension = FALSE,
+                               per = 'fortnight',
+                               overseas_absence = FALSE,
+                               seperated_couple = FALSE){
+  
+  if(!(per %in% c('fortnight', 'annual'))){
+    stop("per can only take values `fortnight` or `annual`")
+  }
+  
+  if(any(!has_partner & seperated_couple)) {
+    stop("incompatible values of `has_partner` and `partner_seperated`")
+  }
+  
+  if (is.null(Date)) {
+    if (is.null(fy.year)) {
+      Date <- as.Date("2016/03/01")
+      fy.year <- "2015-16"
+      warning('`Date` and `fy.year` not set, so using `Date` = "2016/03/01" and `fy.year` = "2015-16"')
+    } else {
+      Date <- fy2date(fy.year)
+      warning('`Date` not set. Using date as defined in fy2date()')
```

Comment:
Should not be a warning. Message only if both are unset; no message if one is set. Only documenting the result is necessary